composer/callbacks/system_metrics_monitor.py

```diff
@@ -19,13 +20,56 @@
 
 __all__ = ['SystemMetricsMonitor']
 
+_GPU_METRICS = [
+   'gpu_percentage',
+   'memory_percentage',
+   'gpu_temperature_C',
+   'gpu_power_usage_W',
+]
+
 
 class SystemMetricsMonitor(Callback):
-    """Track system metrics."""
+    """Logs GPU and CPU metrics relevant to straggler detection across all ranks.
+
+    GPU Metrics:
+        gpu_percentage: Occupancy rate, percent of time over the past sampling period during
+                        which one or more kernels was executing on the GPU.
+        memory_percentage: Percent of time over the past sampling period during which
+                        global (device) memory was being read or written.
+        gpu_temperature_C: Temperature of device, in Celcius.
+        gpu_power_usage_W: Power usage of device, in Watts.
+
+    If the log_all_data flag is set to false (which is false by default), only the maximum and minimum values
+    for these metrics, alongside their respective ranks in the key names, are logged
```

Comment:
since default is false, instead you should say if it's set to `True`, here is additional information thats logged